### DISCUSSION TITLE
image in style inline

### DISCUSSION THREAD

**Discussion by damiankomonski** [webpack/webpack#18521] [Q&A]
hello, how can I load image from html file from style?
`<div style="background-image: url('images/abc123.png')">`

When I write in such way webpack doesnt load image. How can I make it?

I am using html-loader, asset/resource, css-loader, postcss-loader

Please help me, I need it...

**Comment by alexander-akait** [comment]
Using `new URL("./images/abc123.png", import.meta.url)` and include it in your string

**Reply by damiankomonski** [reply]
Should I make it from javascript file? Values will change by PHP.

**Reply by damiankomonski** [reply]
I dont understand "new URL()", please explain

**Reply by alexander-akait** [reply]
> I dont understand "new URL()", please explain

https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

> Should I make it from javascript file? Values will change by PHP.

You can generate JS code on PHP side

**Reply by damiankomonski** [reply]
whether webpack will not see the path in style? 

I rather have to import image firstly.

**Reply by alexander-akait** [reply]
How do you generate `<div style="background-image: url('images/abc123.png')">` in PHP? Show me part of code

**Reply by damiankomonski** [reply]
I code only in HTML. But value can be added from php.

**Reply by damiankomonski** [reply]
it is very annoying that I cant write simply style="background-image....", I have to write javascript code, it is nonsense

**Reply by alexander-akait** [reply]
How we should undestand what is just a string and what is a string with path, so yes, you need to write analizable construction, to avoid such problems, just use classes and CSS, we have an issue - https://github.com/webpack-contrib/html-loader/issues/259

**Reply by damiankomonski** [reply]
Ok thanks for help, i will use css classes in this situation…
